.docker.sh

```diff
@@ -1,30 +1,8 @@
-#!/usr/bin/env bash
-DISTRO=${DISTRO:-alpine}
+#!/bin/bash
+set -e
 
-set -ex
-# TODO opam2 depext
-case $DISTRO in
-alpine*) sudo apk add m4 ;;
-debian*) sudo apt update; sudo apt -y install m4 pkg-config ;;
-ubuntu*) sudo apt update; sudo apt -y install m4 pkg-config ;;
-esac
-
-sudo chown -R opam /home/opam/src
-cd /home/opam/src
-echo "Installing dependencies"
-opam pin add --no-action duniverse .
-opam update
-opam install --deps-only -t duniverse
-opam install ocamlformat=0.9
-echo "Building and running tests"
-make
-make test
-make install
-(cd examples
- ./build-examples.sh)
-echo "Checking code formatting"
-make format
-if ! git diff --exit-code ; then
-  echo "Incorrect formatting, please run 'make format' before comitting changes"
-  exit 1
+if [ -z "$OCAMLFORMAT" ]; then
+  docker build --build-arg DISTRO=$DISTRO -f .dockerfiles/opam .
+else
+  docker build --build-arg OCAMLFORMAT=$OCAMLFORMAT -f .dockerfiles/ocamlformat .
```

Comment:
I'm not fan of this script. It is here for having a clean `matrix` in `.travis.yaml` but it feels a bit magical to me.

What do you think of something like this instead:

```yaml
script: docker build --build-arg "$BUILD_ARG" -f "$DOCKERFILE" .
matrix:
  - BUILD_ARG=DISTRO=distro-1 DOCKERFILE=.dockerfiles/opam
  ...
  - BUILD_ARG=OCAMLFORMAT=0.9 DOCKERFILE=.dockerfiles/ocamlformat
```

Reply:
The build arg part is easy to mess up so I'm not sure this would be an improvement. Is there a way we can improve the script so that it would feel a bit clearer what's happening?

Reply:
Also `--build-arg` only takes one variable, you have to pass it repeatedly to set several variables so that wouldn't be extensible at all.

Reply:
I see. Repeating script commands would be even worse. We could try using the "stage" feature from travis but I don't known how it works.

Reply:
I'm not sure the stage would be much better but it could indeed work. What I like about the current design is you don't have to make an ordering choice between quality and regular builds, which you'd need to do with stages.